lib/_instance.ts

```diff
@@ -0,0 +1,22 @@
+import AlgoliaAnalytics from "./insights";
+import { processQueue } from "./_processQueue";
+import { getFunctionalInterface } from "./_getFunctionalInterface";
+
+export function getInstance() {
+  return new AlgoliaAnalytics();
+}
+
+export function getInstanceForBrowser() {
+  const instance = getInstance();
+  if (typeof window !== "undefined") {
+    // Process queue upon script execution
+    processQueue.call(instance, window);
+  }
+  return instance;
+}
+
+export function getInstanceForNode() {
+  const instance = new AlgoliaAnalytics();
+  const functionalInterface = getFunctionalInterface(instance);
+  return functionalInterface;
+}
```

Comment:
Wouldn't it be simpler if we just put getInstantForNode inside node
and getInstanceForBrowser inside browser.ts ?

Reply:
I tried put `getInstanceForBrowser` inside `browser.ts`. In the `browser.ts`, there were a default export `getInstanceForBrowser()` and a named export `getInstanceForBrowser`.
And it somehow broke the umd build. So I had to get rid of the named export.